src/ServiceControlInstaller.Engine/Configuration/ServiceControl/ServiceControlAppConfig.cs

```diff
@@ -32,6 +32,12 @@ protected override void UpdateSettings()
             settings.Set(ServiceControlSettings.EnableFullTextSearchOnBodies, details.EnableFullTextSearchOnBodies.ToString(), version);
             settings.Set(ServiceControlSettings.RemoteInstances, RemoteInstanceConverter.ToJson(details.RemoteInstances), version);
 
+            // Windows services allow a maximum of 125 seconds when stopping a service.
+            // When shutting down or restarting the OS we have no control over the
+            // shutdown timeout. This is by the installer engine that is run _only_ on
+            // Windows via SCMU or PowerShell
+            settings.Set(ServiceControlSettings.ShutdownTimeout, "00:02:00");
```

Comment:
Not that familiar with this. This is added during an upgrade to the config?

Reply:
Either way, upgrade or new instances